service/kvmclusterconfig/framework.go

```diff
@@ -11,22 +11,27 @@ import (
 	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
 	"k8s.io/client-go/kubernetes"
 
+	"github.com/giantswarm/cluster-operator/pkg/cluster"
 	"github.com/giantswarm/cluster-operator/service/kvmclusterconfig/v1"
 )
 
 // FrameworkConfig contains necessary dependencies and settings for
 // KVMClusterConfig CRD framework implementation.
 type FrameworkConfig struct {
-	G8sClient    versioned.Interface
-	K8sClient    kubernetes.Interface
-	K8sExtClient apiextensionsclient.Interface
-	Logger       micrologger.Logger
+	BaseClusterConfig *cluster.Config
+	G8sClient         versioned.Interface
+	K8sClient         kubernetes.Interface
+	K8sExtClient      apiextensionsclient.Interface
+	Logger            micrologger.Logger
 
 	ProjectName string
 }
 
 // NewFramework returns a configured KVMClusterConfig framework implementation.
 func NewFramework(config FrameworkConfig) (*framework.Framework, error) {
+	if config.BaseClusterConfig == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.BaseClusterConfig must not be empty")
+	}
```

Comment:
IMO this should not be validated here because it is not used. I do not even see it being forwarded. So this property should maybe not be here at all.

Reply:
It will be forwarded once `certconfig` is registered to resourceset. Wouldn't validation then make sense?

Reply:
When it is only forwarded from this layer here to another layer then we are not making use of the thing and thus do not need to care since the layer that receives and actually uses it should only validate because it wants to use it. This here is only a wire.